Allow AssertOutOfBoundsWrapper to be applied to any environment

Bamboofungus opened the pull request:
# Description

Allows `AssertOutOfBoundsWrapper` to be applied to environments with any action space type, resolving https://github.com/Farama-Foundation/PettingZoo/issues/866.

## Type of change

- New feature (non-breaking change which adds functionality)

### Screenshots

> Please attach before and after screenshots of the change if applicable.
> To upload images to a PR -- simply drag and drop or copy paste.

# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [x] I have run `pytest -v` and no errors are present.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I solved any possible warnings that `pytest -v` has generated that are related to my code to the best of my knowledge.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

elliottower commented:
It turns out that all the spaces have `.contains()` now so this can be applied to any type of space, if you could update it accordingly that would be great. Another thing is we could probably implement something like https://github.com/Farama-Foundation/Gymnasium/blob/main/gymnasium/utils/passive_env_checker.py or take some of the code from it, but I think that should be a new PR

pseudo-rnd-thoughts reviewed (changes requested):
@elliottower why can this wrapper not be made general such that it works with all spaces?

elliottower commented:
Yea that’s the plan, jet and I discussed a bit in the  discord if you
didn’t see

On Tue, Jul 25, 2023 at 5:29 PM Mark Towers ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> @elliottower <https://github.com/elliottower> why can this wrapper not be
> made general such that it works with all spaces?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Farama-Foundation/PettingZoo/pull/1046#pullrequestreview-1546491779>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHVPVA57I5HXM2TTDPK6BITXSA3CZANCNFSM6AAAAAA2XNFI3I>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Bamboofungus commented:
@pseudo-rnd-thoughts I removed the check for Discrete environments and changed the unwrapped tests so that the wrapper gets tested on all space types, can you rereview?

pseudo-rnd-thoughts reviewed (approved):
Thanks for the changes, looks good to me
